### ISSUE TITLE
Integrate new engine icon settings

### ISSUE THREAD

**Issue by GoogleFrog** [ZeroK-RTS/Zero-K#4809]
Supcom-like icons exists with the following settings.

**Comment by sprunk** [comment]
Kinda related: #2880

As a vet I'm used to icons always being clickable and I kept right-clicking ghost icons.

**Comment by GoogleFrog** [comment]
Are you saying that the new engine icons have ghosts?

**Comment by sprunk** [comment]
Yes, or rather, ghosts have new engine icons, and this happens even outside radar (they're also visible as icons on the minimap but that is less of a problem). The new engine icons render on ghosts in the same way they render on units visible to radar, which involves the same problem as #2880 where they're hard to distinguish and people might right-click them. Possibly that issue is even worse here because ghost models look different compared to visible models (due to transparency, lack of shaders and animation), while the ghost icons are (as far as I can tell) exactly the same as regular radar icons.

**Comment by GoogleFrog** [comment]
Make a feature request to toggle this?

**Comment by sprunk** [comment]
The icon turns out to be los hax, and I think ghosts may have been subtly los hax all along (unless something changed recently). It is horrifying. https://github.com/beyond-all-reason/spring/issues/640

**Comment by GoogleFrog** [comment]
Ghosts still get icons when they are alive.
![image](https://github.com/ZeroK-RTS/Zero-K/assets/1318249/f3af2815-ecc5-4e7b-bfa2-83bcf284150f)
![image](https://github.com/ZeroK-RTS/Zero-K/assets/1318249/1f6e76c2-c829-418b-9e4f-6210df67370d)

**Comment by GoogleFrog** [comment]
Fade feels much to gradual to me. I want a way to configure it.

The lower end is ok:
![image](https://github.com/ZeroK-RTS/Zero-K/assets/1318249/33239ede-8785-4096-bd5f-d1970f4bb688)
![image](https://github.com/ZeroK-RTS/Zero-K/assets/1318249/c28b4f82-d46c-4c13-8eeb-0da07e3557e3)

At this zoom level I would want icons looking quite similar to how they look now:
![image](https://github.com/ZeroK-RTS/Zero-K/assets/1318249/a88cc0c3-e5fc-4fdf-82e3-16608bcde103)
![image](https://github.com/ZeroK-RTS/Zero-K/assets/1318249/5489405c-ff90-4ab3-8207-c163dba02ad9)

I think the setting is more for smoothing an otherwise sharp transition, than for being something people see most of the time. Having a transparent draw over a unit for the majority of zoom levels is a bit messy.

**Comment by GoogleFrog** [comment]
Also, icons should be drawn based on the distance between the camera and the unit, rather than how it happens now with a global distance factor.

**Comment by GoogleFrog** [comment]
Iconsasui also seems to ignore zoom distance threshold settings for icons.
![image](https://github.com/ZeroK-RTS/Zero-K/assets/1318249/6c69bee8-7e25-4c87-9dce-246cb3262355)
![image](https://github.com/ZeroK-RTS/Zero-K/assets/1318249/eda97906-e440-4c60-ace2-016282c1b130)

**Comment by sprunk** [comment]
There was talk of stripping this mode from the engine and making a GL4 widget. Presumably this would make it naturally obey ghost/LoS rules and also be more easily configurable.

**Comment by GoogleFrog** [comment]
That might work. Also how does it interact with selection volumes, and how would a widget version do that? How would a widget tell the models not to draw? Seems tricky or at least expensive to turn it into a widget with the required amount of polish.

**Comment by lhog** [comment]
> Ghosts still get icons when they are alive.

Is area covered by a friendly radar?

**Comment by GoogleFrog** [comment]
Another fatal issue is that the radar icons are tiny when zoomed in.
![image](https://github.com/ZeroK-RTS/Zero-K/assets/1318249/7faa383f-6ab9-4b9f-be19-d6049cd59c87)

Old system:
![image](https://github.com/ZeroK-RTS/Zero-K/assets/1318249/46856122-012f-4618-900e-58e1618f049a)
